File: eMovieFinder/eMovieFinder.Model/Dtos/Requests/Director/DirectorInsertRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace eMovieFinder.Model.Dtos.Requests.Director
{
    public class DirectorInsertRequest
    {
        [Required(ErrorMessage = "FirstName is a Required field")]
        public string FirstName { get; set; }
        [Required(ErrorMessage = "LastName is a Required field")]
        public string LastName { get; set; }
        [Required(ErrorMessage = "BirthDate is a Required field")]
        public DateTime BirthDate { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/dtos/requests/director/director_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'director_insert_request.g.dart';

@JsonSerializable()
class DirectorInsertRequest {
  final String firstName;
  final String lastName;
  final DateTime? birthDate;

  DirectorInsertRequest({
    required this.firstName,
    required this.lastName,
    this.birthDate
  });

  factory DirectorInsertRequest.fromJson(Map<String, dynamic> json) => _$DirectorInsertRequestFromJson(json);
  Map<String, dynamic> toJson() => _$DirectorInsertRequestToJson(this);
}